reference/1_application_flow/1_3_synchronization/README.md

```diff
@@ -0,0 +1,140 @@
+# Synchronization
+
+Cranq offers the following synchronization primitives to manage application flow:
+
+### Syncer & splitter
+
+The ```flow/Syncer``` aggregates the input signals & forwards it only, once all of it's inputs received a trigger. 
+- It's counterpart is ```flow/Splitter```, which is capable of splitting up the aggregated signal to the original signals.
+- They both have dynamic input/output ports, allowing the developer to scale them as needed
+- They are useful for building a synchronization point on node inputs, making sure the node executes only once all (or some) of it's inputs are fulfilled
+- They can also be used for merging & splitting data structures, as [demonstrated later](cranqj/app_flow_sync_split.cranqj)
+
+### Multiplexing & demultiplexing
+
+The ```flow/Multiplexer``` node also combines several inputs into a single output, without synchronization
+- The input is forwarded immediately, retaining the input port the original signal was received on. 
+- It's counterpart is ```flow/Demultiplexer```
+- They both have dynamic input/output ports, allowing the developer to scale them as needed
+
+### Repeater
+
+The ```flow/Repeater``` node duplicates the signal it receives, assigning unique tags to each.
+- It assigns a unique, sequentially indexed tag to each duplicate

```

Comment:
I think "sequentially indexed" is an implementation detail, not sure whether we should state it in the documentation.

Reply:
It's important in order to understand why the repeater is needed. Until we figure something out to make the usage of iterator derivatives free of this requirement, I think it is fair to explain why it is like so